mmv1/third_party/terraform/resources/resource_apigee_shared_flow.go

```diff
@@ -0,0 +1,439 @@
+				Computed:    true,
+				Description: `The id of the most recently created revision for this shared flow.`,
+			},
+			"meta_data": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: `Metadata describing the shared flow.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"created_at": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `Time at which the API proxy was created, in milliseconds since epoch.`,
+						},
+						"last_modified_at": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `Time at which the API proxy was most recently modified, in milliseconds since epoch.`,
+						},
+						"sub_type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The type of entity described`,
+						},
+					},
+				},
+			},
+			"revision": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: `A list of revisions of this shared flow.`,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"config_bundle": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `A path to the config bundle zip you want to upload. Must be defined if content is not.`,
+			},
+			"md5hash": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Base 64 MD5 hash of the uploaded config bundle.`,
+			},
+			"detect_md5hash": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "Different Hash",
+				Description: `A hash of local config bundle in string, user needs to use a Terraform Hash function of their choice. A change in hash will trigger an update.`,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					localMd5Hash := ""
+					if config_bundle, ok := d.GetOkExists("config_bundle"); ok {
+						localMd5Hash = getFileMd5Hash(config_bundle.(string))
+					}
+					if localMd5Hash == "" {
+						return false
+					}
+
+					// `old` is the md5 hash we speculated from server responses,
+					// when apply responded with succeed, hash is set to the hash of uploaded bundle
+					if old != localMd5Hash {
+						return false
+					}
+
+					return true
+				},
+			},
+		},
+		UseJSONNumber: true,
+	}
+}
+
+func resourceApigeeSharedFlowCreate(d *schema.ResourceData, meta interface{}) error {
+	ctx := context.TODO()
+	tflog.Info(ctx, "resourceApigeeSharedFlowCreate")
+	log.Printf("[DEBUG] resourceApigeeSharedFlowCreate")
+
+	//print debug info======
+	tmpName, _ := d.GetOk("name")
+	tmpOrgId, _ := d.GetOk("org_id")
+	tmpBundle, _ := d.GetOk("config_bundle")
+	
+	log.Printf("[DEBUG] resourceApigeeSharedFlowCreate, name=			 	%s", tmpName)
+	log.Printf("[DEBUG] resourceApigeeSharedFlowCreate, org_id=, 			%s", tmpOrgId)
+	log.Printf("[DEBUG] resourceApigeeSharedFlowCreate, config_bundle=, 	%s", tmpBundle)
+	//debug=================
```

Comment:
nit: are we able to make lines 134 -144 a one liner in the debug log? and i'd say, if we're going to ignore the errors on lines 137-139, we might as well just do `d.Get("name").(string)`.